### PR TITLE
Gradient accumulation for TFTrainer

### PR DESCRIPTION
**Pull request by kiyoungkim1**
# What does this PR do?
```TFTrainer``` does not work with ```gradient_accumulation_steps``` > 1 (I am doing with ```TFGPT2LMHeadModel```).
Similar treatment of #6479 is done for labels.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors which may be interested in your PR.

 tensorflow: @jplu

### CODE REVIEW

**Review by jplu — APPROVED** [review]
Thanks for the fix! LGTM

File: src/transformers/trainer_tf.py [hunk 1]
```diff
@@ -638,7 +638,9 @@ def apply_gradients(self, features, labels, nb_instances_in_global_batch):
                 reduced_features = {
                     k: ft[: self.args.train_batch_size // self.args.n_replicas] for k, ft in features.items()
                 }
-                reduced_labels = labels[: self.args.train_batch_size // self.args.n_replicas]
+                reduced_labels = {
+                    k: ft[: self.args.train_batch_size // self.args.n_replicas] for k, ft in labels.items()
```
Comment (sgugger) [inline src/transformers/trainer_tf.py:642]:
```suggestion
                    k: lbl[: self.args.train_batch_size // self.args.n_replicas] for k, lbl in labels.items()
```
ft was an abbreviation of feature in the code you pasted, it makes less sense here.

File: src/transformers/trainer_tf.py [hunk 2]
```diff
@@ -650,9 +652,13 @@ def apply_gradients(self, features, labels, nb_instances_in_global_batch):
...
 
-                labels = tf.concat(
-                    [labels[self.args.train_batch_size // self.args.n_replicas :], reduced_labels], axis=0
-                )
+                labels = {
+                    k: tf.concat(
+                        [ft[self.args.train_batch_size // self.args.n_replicas :], reduced_labels[k]],
+                        axis=0,
+                    )
+                    for k, ft in labels.items()
```
Comment (sgugger) [inline src/transformers/trainer_tf.py:660]:
```suggestion
                        [lbl[self.args.train_batch_size // self.args.n_replicas :], reduced_labels[k]],
                        axis=0,
                    )
                    for k, lbl in labels.items()
```

**Review by sgugger — APPROVED** [review]
Thanks for fixing, just left some nits.

**Comment by jplu** [comment]
Sorry we have to revert this PR as `labels` is not a dict but a tensor.

**Comment by jplu** [comment]
Thanks a lot for having spotted this cas, a more adapted fix will be available here #9616 very sorry for the inconvenience.

**Comment by kiyoungkim1** [comment]
Never mind. Thanks again for the fix.
